qiskit_ibm_provider/job/ibm_circuit_job.py

```diff
@@ -595,56 +523,36 @@ def refresh(self) -> None:
             IBMJobApiError: If an unexpected error occurred when communicating
                 with the server.
         """
+        # TODO: Change to use runtime response data as much as possible
         with api_to_job_error():
-            api_response = self._api_client.job_get(self.job_id())
+            api_response = self._runtime_client.job_get(self.job_id())
+            api_metadata = self._runtime_client.job_metadata(self.job_id())
 
         try:
-            api_response.pop("job_id")
-            self._creation_date = dateutil.parser.isoparse(
-                api_response.pop("creation_date")
-            )
-            self._api_status = api_response.pop("status")
-            if "kind" in api_response:
-                self._kind = ApiJobKind(api_response.pop("kind"))
-            if "qobj" in api_response:
-                self._qobj = dict_to_qobj(api_response.pop("qobj"))
+            api_response.pop("id")
+            self._creation_date = dateutil.parser.isoparse(api_response.pop("created"))
+            self._api_status = api_response.pop("state")["status"]
         except (KeyError, TypeError) as err:
             raise IBMJobApiError(
                 "Unexpected return value received " "from the server: {}".format(err)
             ) from err
 
-        self._name = api_response.pop("name", None)
         self._time_per_step = api_response.pop("time_per_step", None)
```

Comment:
`time_per_step` is actually `timestamps` from job metrics (see [here](https://runtime-us-east.quantum-computing.ibm.com/openapi/#/Jobs/get_job_metrics_jid))